### PR TITLE
The indexes of DataFrame.describe(percentiles=[0.29, 0.57, 0.58]) are incorrect

### LINKED ISSUE #46362: BUG: The indexes of `DataFrame.describe(percentiles=[0.29, 0.57, 0.58])` are incorrect
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58])

 	value
count 	3.00
mean 	2.00
std 	1.00
min 	1.00
29.0% 	1.58
50% 	2.00
57.0% 	2.14
58.0% 	2.16
max 	3.00
```

### Issue Description

The indexes ['29.0%', '57.0%', '58.0%'] should be named as ['29%', '57%', '58%'].

### Expected Behavior

```
pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58]).index

# Output:
Index(['count', 'mean', 'std', 'min', '29.0%', '50%', '57.0%', '58.0%', 'max'], dtype='object')

# Expected:
Index(['count', 'mean', 'std', 'min', '29%', '50%', '57%', '58%', 'max'], dtype='object')
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 06d230151e6f18fdb8139d09abf539867a8cd481
python           : 3.9.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.13.0-30-generic
Version          : #33-Ubuntu SMP Fri Feb 4 17:03:31 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_IL
LOCALE           : en_IL.UTF-8

pandas           : 1.4.1
numpy            : 1.22.2
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 22.0.3
setuptools       : 59.8.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : 4.4.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.8.0
html5lib         : None
pymysql          : None
psycopg2         : 2.9.3
jinja2           : 3.0.3
IPython          : 7.32.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : 2022.02.0
gcsfs            : None
matplotlib       : 3.5.1
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.02.0
scipy            : 1.8.0
sqlalchemy       : 1.4.31
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by weikhor**
- [ ] closes #46362
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/formats/format.py [hunk 1]
```diff
@@ -1721,10 +1721,10 @@ def format_percentiles(
 
     percentiles = 100 * percentiles
 
-    int_idx = np.isclose(percentiles.astype(int), percentiles)
+    int_idx = np.isclose(percentiles.round().astype(int), percentiles)
```
Comment (mroeschke) [inline pandas/io/formats/format.py:1724]: Could you make `percentiles.round().astype(int)` a dedicated variable so it can be used here and below?

**Review by mroeschke — COMMENTED** [review]
Looks fairly good. Could use a whatsnew note in `1.6.rst`

File: doc/source/whatsnew/v1.6.0.rst [hunk 2]
```diff
@@ -151,7 +151,7 @@ Interval
 
 Indexing
 ^^^^^^^^
--
+- Bug in :func:`format_percentiles` when formatting percentiles, more decimals than needed shown(:issue:`46362`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.6.0.rst:154]:
```suggestion
- Bug in :func:`~DataFrame.describe` when formatting percentiles in the resulting index showed more decimals than needed (:issue:`46362`)
```

Comment (weikhor) [reply doc/source/whatsnew/v1.6.0.rst]: ok

File: pandas/io/formats/format.py [hunk 3]
```diff
@@ -1720,11 +1720,12 @@ def format_percentiles(
...
 
     percentiles = 100 * percentiles
+    percentiles_round_type = percentiles.round().astype(int)
 
-    int_idx = np.isclose(percentiles.astype(int), percentiles)
+    int_idx = np.isclose(percentiles_round_type, percentiles)
 
     if np.all(int_idx):
-        out = percentiles.astype(int).astype(str)
+        out = percentiles_round_type.astype(str)
```
Comment (mroeschke) [inline pandas/io/formats/format.py:1728]: Does the example in the original issue hit this path or the one below? Just want to make sure we have 2 tests that didn't work before and hit this change and the one below to work correctly now

Comment (weikhor) [reply pandas/io/formats/format.py:1728]: Hi, Does the example in the original issue hit this path or the one below?  

a) The example from issue will hit this path.
```
pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58])
```
```
if np.all(int_idx):
        out = percentiles_round_type.astype(str)
        return [i + "%" for i in out]
```
b) The example below I own create will hit below path.
```
pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.291, 0.57, 0.58])
```
```
out[int_idx] = percentiles[int_idx].round().astype(int).astype(str)
```

Comment (mroeschke) [reply pandas/io/formats/format.py:1728]: Could you add the `pd.DataFrame(data=[1, 2, 3], columns=['value']).describe(percentiles=[0.29, 0.57, 0.58])` case as a unit test?

Comment (weikhor) [reply pandas/io/formats/format.py:1728]: Have add test

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @weikhor!

**Pull request merged**